File: Proyecto/Backend/Spring/hogar/src/main/java/com/roomiegram/hogar/model/Visita.java
package com.roomiegram.hogar.model;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "visitas")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Visita {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "hogar_id", nullable = false)
    private Long hogarId;

    @Column(name = "usuario_visitante_id", nullable = false)
    private Long usuarioVisitanteId;

    @Column(name = "fecha_visita", nullable = false)
    private LocalDateTime fechaVisita;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private EstadoVisita estado = EstadoVisita.PENDIENTE;

    @Column(length = 1000)
    private String comentarios;

    @Column(name = "fecha_creacion", nullable = false)
    private LocalDateTime fechaCreacion;

    @PrePersist
    public void prepararNuevaVisita() {
        if (fechaCreacion == null) {
            fechaCreacion = LocalDateTime.now();
        }
        if (estado == null) {
            estado = EstadoVisita.PENDIENTE;
        }
    }
}

File: Proyecto/Frontend/Roomiegram/src/types/Visita.ts
export type EstadoVisita = "PENDIENTE" | "REALIZADA" | "CANCELADA"

export type Visita = {
  id: number
  hogarId: number
  usuarioVisitanteId: number
  fechaVisita: string
  estado: EstadoVisita
  comentarios: string | null
  fechaCreacion: string
}

export type CrearVisitaPayload = {
  usuarioVisitanteId: number
  fechaVisita: string
  comentarios?: string
}

export type ActualizarVisitaPayload = {
  estado: EstadoVisita
  comentarios?: string
  administradorId: number
}
